### PR TITLE
Add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by ticoniq**
**### Add Person, Student, and Teacher classes**

This pull request addresses the addition of essential classes to the project: Person, Student, and Teacher. These classes lay the foundation for the user hierarchy within the system.

Changes Made:

Person Class:
Student Class:
Teacher Class:

### CODE REVIEW

File: classes/student.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Student < Person
+  def initialize(classroom)
```
Comment (Davidosky007) [inline classes/student.rb:4]: *  This constructor is meant to have all the parameters of Person class too and by inclusion, classroom, Kindly add them as its [required](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_people.md#:~:text=Create%20class%20Student,%22%C2%AF%5C(%E3%83%84)/%C2%AF%22.)  HINTS 💭 `def initialize(age, classroom, name = "Unknown", parent_permission = true)  `

File: classes/teacher.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Teacher < Person
+  def initialize(specialization)
```
Comment (Davidosky007) [inline classes/teacher.rb:4]: *  This constructor is meant to have all the parameters of Person class too and by inclusion, classroom, Kindly add them as its [required](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_people.md#:~:text=Create%20class%20Student,%22%C2%AF%5C(%E3%83%84)/%C2%AF%22.)  HINTS 💭
 ```ruby
def initialize(age, specialization, name = "Unknown", parent_permission = true)
    super(age, name, parent_permission)
```

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @ticoniq 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* Well written PR 👍 
* Classes and Objects in ruby implemented properly 👍 
* No linter errors 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by roshan-bajgain — APPROVED** [review]
Hi @ticoniq, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors. ✔️
- Gitflow is followed. ✔️
- Good job on implementing the changes ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**